mono/mini/debugger-agent.c

```diff
@@ -5148,6 +5148,10 @@ do_invoke_method (DebuggerTlsData *tls, Buffer *buf, InvokeData *invoke)
 
 	if (m->is_generic && !m->is_inflated)
 		return ERR_NOT_IMPLEMENTED;
+	
+	/* Do not invoke method with a generic type parameter as return value. */
+	if(sig && sig->ret && sig->ret->type == MONO_TYPE_VAR)
```

Comment:
We can get in this situation with VSTU when using hovering over Value in a line like this:

`Singleton<TestSingleton>.Instance.Value = 1;`

This does not happen when using evaluating the expression in the immediate window, so I think it is an issue with the hovering performing a bad eval request which crashes the debugger.